Fix README

kitmonisit opened the pull request:
Fix markdown for code blocks and variable names

Add examples for `g:verilog_syntax_fold`

vhda commented:
I like some of your suggestions, while others not so much. Would you mind if I change the README file on my side first and then ask you to rebase your changes over the newer version?

In particular, why modify the ````` code blocks to the indented format? Are you using any specific parser that does not support it?

kitmonisit commented:
I'm not sure which parts you don't like. Please go ahead and make changes, and we can discuss it.

In particular, I've always known that backticks are used for inline code. However, in the original README, they looked like they were intended as code blocks, which are denoted by 4 spaces. Unless the triple backtick is a new specification, I haven't found any definition of it. A quick search gets me this: [What is the ``` (3 x backquote) markdown used for?](http://meta.stackexchange.com/a/189921/150078)

vhda commented:
I admit that I've mostly been using GitHub Flavored Markdown, where the triple backquote is used to define [Fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks). One of the advantages of this format is that it allows for [Syntax highlighting](https://help.github.com/articles/github-flavored-markdown/#syntax-highlighting), even though I've not been using it much.

I've introduced the use of this feature in the `doc/readme` branch and already added/adapted most of your suggestions.

kitmonisit commented:
I rebased off your `doc/readme` branch and added in my changes. I also added examples for `g:verilog_syntax_fold` and updated its instructions. Previously, it was

> Comma separated list ...

Strictly speaking, `g:verilog_syntax_fold` does not look for a `list`, but a `string`, so I updated the instructions to make it more clear.

kitmonisit commented:
All right, done :)

vhda commented:
Could I kindly ask you to rebase your changes over master again?
I can find an old commit of mine along side with yours when I try to merge your PR.

kitmonisit commented:
Sure, I just finished rebasing it as requested :)

vhda commented:
Thank you very much for your contribution :)
